.github/actions/build-image/action.yml

```diff
@@ -54,11 +57,15 @@ runs:
 
         LAST_BRANCH_BUILD="$REMOTE_DOCKER_REPOSITORY:${{ inputs.branch-name }}"
         if ! docker pull "$LAST_BRANCH_BUILD"; then
-          LAST_MASTER_BUILD="$REMOTE_DOCKER_REPOSITORY:${{ inputs.branch-main }}"
-          echo "Could not pull $LAST_MASTER_BUILD, will try last master build $LAST_MASTER_BUILD"
-          if ! docker pull "$LAST_MASTER_BUILD"; then
-            echo "Could not pull $LAST_MASTER_BUILD either ¯\_(ツ)_/¯. Will just have to build the image from scratch."
-          fi;
+          if [ "${{ inputs.skip-build-optimization }}" == "false" ]; then
+            LAST_MASTER_BUILD="$REMOTE_DOCKER_REPOSITORY:${{ inputs.branch-main }}"
+            echo "Could not pull $LAST_BRANCH_BUILD, will try last master build $LAST_MASTER_BUILD"
+            if ! docker pull "$LAST_MASTER_BUILD"; then
```

Comment:
Not actionable, just thinking that this is quite inefficient, likely pulling some layers unnecessarily, while it should be easy to check whether a specific layer is present in the registry / remote cache at build time...

Reply:
Maybe possible if the entire approach is re-thought, as the cached images are now compressed, so the only way right now to reuse their layers is to load them up locally first. But I get your point